agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/AgentLogExporter.java

```diff
@@ -0,0 +1,110 @@
+
+import com.azure.core.util.CoreUtils;
+import com.azure.monitor.opentelemetry.exporter.implementation.LogDataMapper;
+import com.azure.monitor.opentelemetry.exporter.implementation.logging.OperationLogger;
+import com.azure.monitor.opentelemetry.exporter.implementation.models.TelemetryItem;
+import com.azure.monitor.opentelemetry.exporter.implementation.quickpulse.QuickPulse;
+import com.microsoft.applicationinsights.agent.internal.telemetry.BatchItemProcessor;
+import com.microsoft.applicationinsights.agent.internal.telemetry.TelemetryClient;
+import com.microsoft.applicationinsights.agent.internal.telemetry.TelemetryObservers;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.logs.data.LogData;
+import io.opentelemetry.sdk.logs.data.Severity;
+import io.opentelemetry.sdk.logs.export.LogExporter;
+import java.util.Collection;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AgentLogExporter implements LogExporter {
+
+  private static final Logger logger = LoggerFactory.getLogger(AgentLogExporter.class);
+
+  private static final OperationLogger exportingLogLogger =
+      new OperationLogger(AgentLogExporter.class, "Exporting log");
+
+  // TODO (trask) could implement this in a filtering LogExporter instead
+  private volatile Severity threshold;
+
+  private final LogDataMapper mapper;
+  private final Consumer<TelemetryItem> telemetryItemConsumer;
+
+  public AgentLogExporter(
+      Severity threshold,
+      LogDataMapper mapper,
+      @Nullable QuickPulse quickPulse,
+      BatchItemProcessor batchItemProcessor) {
+    this.threshold = threshold;
+    this.mapper = mapper;
+    telemetryItemConsumer =
+        telemetryItem -> {
+          if (quickPulse != null) {
+            quickPulse.add(telemetryItem);
+          }
+          TelemetryObservers.INSTANCE
+              .getObservers()
+              .forEach(consumer -> consumer.accept(telemetryItem));
+          batchItemProcessor.trackAsync(telemetryItem);
+        };
+  }
+
+  public void setThreshold(Severity threshold) {
+    this.threshold = threshold;
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<LogData> logs) {
+    if (CoreUtils.isNullOrEmpty(TelemetryClient.getActive().getInstrumentationKey())) {
+      logger.debug("Instrumentation key is null or empty. Fail to export logs.");
+      return CompletableResultCode.ofFailure();
+    }
+    for (LogData log : logs) {
+      logger.debug("exporting log: {}", log);
+      try {
+        int severity = log.getSeverity().getSeverityNumber();
+        int threshold = this.threshold.getSeverityNumber();
+        if (severity < threshold) {
+          continue;
+        }
+        mapper.map(log, telemetryItemConsumer);
+        exportingLogLogger.recordSuccess();
+      } catch (Throwable t) {
+        exportingLogLogger.recordFailure(t.getMessage(), t);
```

Comment:
return ofFailure right after this?

Reply:
just returning success for all "Agent" exporters since all the failure behavior is handled internally, I'll add a comment to explain better